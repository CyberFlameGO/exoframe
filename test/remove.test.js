/* eslint-env jest */
// mock config for testing
jest.mock('../src/config', () => require('./__mocks__/config'));

// npm packages
const nock = require('nock');
const sinon = require('sinon');

// our packages
const {handler: remove} = require('../src/commands/remove');
const cfg = require('../src/config');

const id = 'test-id';
const url = 'test.example.com';

// test removal
test('Should remove', done => {
  // handle correct request
  const rmServer = nock('http://localhost:8080').post(`/remove/${id}`).reply(204);
  // spy on console
  const consoleSpy = sinon.spy(console, 'log');
  // execute login
  remove({id}).then(() => {
    // make sure log in was successful
    // check that server was called
    expect(rmServer.isDone()).toBeTruthy();
    // first check console output
    expect(consoleSpy.args).toMatchSnapshot();
    // restore console
    console.log.restore();
    rmServer.done();
    done();
  });
});

test('Should remove by url', done => {
  const rmServer = nock('http://localhost:8080').post(`/remove/${url}`).reply(204);

  const consoleSpy = sinon.spy(console, 'log');

  remove({id: url}).then(() => {
    // make sure log in was successful
    // check that server was called
    expect(rmServer.isDone()).toBeTruthy();
    // first check console output
    expect(consoleSpy.args).toMatchSnapshot();
    // restore console
    console.log.restore();
    rmServer.done();
    done();
  });
});

test('Should remove by token instead of default auth', done => {
  const rmServer = nock('http://localhost:8080').post(`/remove/${id}`).reply(204);

  const consoleSpy = sinon.spy(console, 'log');

  remove({id: id, token: 'test-token'}).then(() => {
    // make sure log in was successful
    // check that server was called
    expect(rmServer.isDone()).toBeTruthy();
    // first check console output
    expect(consoleSpy.args).toMatchSnapshot();
    // restore console
    console.log.restore();
    rmServer.done();
    done();
  });
});

// test removal error
test('Should show remove error', done => {
  // handle correct request
  const rmServer = nock('http://localhost:8080').post(`/remove/${id}`).reply(500);
  // spy on console
  const consoleSpy = sinon.spy(console, 'log');
  // execute login
  remove({id}).then(() => {
    // make sure log in was successful
    // check that server was called
    expect(rmServer.isDone()).toBeTruthy();
    // first check console output
    expect(consoleSpy.args).toMatchSnapshot();
    // restore console
    console.log.restore();
    rmServer.done();
    done();
  });
});

// test removal error
test('Should show not found error', done => {
  // handle correct request
  const rmServer = nock('http://localhost:8080').post(`/remove/${id}`).reply(404);
  // spy on console
  const consoleSpy = sinon.spy(console, 'log');
  // execute login
  remove({id}).then(() => {
    // make sure log in was successful
    // check that server was called
    expect(rmServer.isDone()).toBeTruthy();
    // first check console output
    expect(consoleSpy.args).toMatchSnapshot();
    // restore console
    console.log.restore();
    rmServer.done();
    done();
  });
});

// test removal error on incorrect success code
test('Should show not found error', done => {
  // handle correct request
  const rmServer = nock('http://localhost:8080').post(`/remove/${id}`).reply(200);
  // spy on console
  const consoleSpy = sinon.spy(console, 'log');
  // execute login
  remove({id}).then(() => {
    // make sure log in was successful
    // check that server was called
    expect(rmServer.isDone()).toBeTruthy();
    // first check console output
    expect(consoleSpy.args).toMatchSnapshot();
    // restore console
    console.log.restore();
    rmServer.done();
    done();
  });
});

// test
test('Should deauth on 401', done => {
  // handle correct request
  const rmServer = nock('http://localhost:8080').post(`/remove/${id}`).reply(401);
  // spy on console
  const consoleSpy = sinon.spy(console, 'log');
  // execute login
  remove({id}).then(() => {
    // make sure log in was successful
    // check that server was called
    expect(rmServer.isDone()).toBeTruthy();
    // first check console output
    expect(consoleSpy.args).toMatchSnapshot();
    // check config
    expect(cfg.userConfig.user).toBeUndefined();
    expect(cfg.userConfig.token).toBeUndefined();
    // restore console
    console.log.restore();
    // tear down nock
    rmServer.done();
    done();
  });
});
