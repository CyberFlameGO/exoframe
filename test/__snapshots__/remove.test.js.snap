// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should deauth on 401 1`] = `
Array [
  Array [
    "[1mRemoving deployment:[22m",
    "test-id",
  ],
  Array [
    "[31mError: authorization expired![39m",
    "Please, relogin and try again.",
  ],
]
`;

exports[`Should remove 1`] = `
Array [
  Array [
    "[1mRemoving deployment:[22m",
    "test-id",
  ],
  Array [
    "[32mDeployment removed![39m",
  ],
]
`;

exports[`Should show not found error 1`] = `
Array [
  Array [
    "[1mRemoving deployment:[22m",
    "test-id",
  ],
  Array [
    "[31mError: container or function was not found![39m",
    "Please, check deployment ID and try again.",
  ],
]
`;

exports[`Should show not found error 2`] = `
Array [
  Array [
    "[1mRemoving deployment:[22m",
    "test-id",
  ],
  Array [
    "[31mError![39m",
    "Could not remove the deployment.",
  ],
]
`;

exports[`Should show remove error 1`] = `
Array [
  Array [
    "[1mRemoving deployment:[22m",
    "test-id",
  ],
  Array [
    "[31mError removing project:[39m",
    "HTTPError: Response code 500 (Internal Server Error)",
  ],
]
`;
